lib/word_counter.rb

```diff
@@ -14,14 +14,22 @@ def initialize(text)
   # scan by word boundaries after stripping hyphens and apostrophes
   # so one-word and one's will be counted as one word, not two.
   # -- is replaced by — (emdash) before strip so one--two will count as 2
+  #
+  # Scripts such as Chinese and Japanese that do not have space between words
+  # are counted based on the number of characters. If a text include mixed
+  # languages, only characters in these languages would be counted as words,
+  # words in other languages are counted normally
   def count
     count = 0
     # avoid blank? so we don't need to load Rails for tests
     return count if @text.nil? || @text.empty?
     body = Nokogiri::HTML(@text).xpath('//body').first
     body.traverse do |node|
       if node.is_a? Nokogiri::XML::Text
-        count += node.inner_text.gsub(/--/, "—").gsub(/['’‘-]/, "").scan(/[[:word:]]+/).size
+        # includes Chinese, Japanese and Thai
+        languages_without_spaces = "\\p{Han}|\\p{Katakana}|\\p{Hiragana}|\\p{Thai}"
```

Comment:
We should move this to the configuration file, so we can add more scripts without deploying code changes. In config/config.yml, if you add:

```yml
# List of Unicode scripts where spaces are not used as separators, and
# character counts should be used for word counts.
# Check if a script is supported by the version of Ruby we use:
# https://ruby-doc.org/core/Regexp.html#class-Regexp-label-Character+Properties
CHARACTER_COUNT_SCRIPTS: ["Han", "Hiragana", "Katakana", "Thai"]
```

then you can get the list here by using `ArchiveConfig.CHARACTER_COUNT_SCRIPTS`.

Reply:
Yes, I'm yj77. I would really appreciate it if you give me the permission on JIRA! And sorry for not reading the testing heading carefully. There are no testing instructions I wish to add.

I've changed the mode of the file back and moved the language list to config. Thank you for the review!